bluebottle/bb_accounts/utils.py

```diff
@@ -20,15 +22,26 @@ def send_welcome_mail(user=None):
         'site_name': tenant_name(),
         'user': user,
         'first_name': user.first_name,
+        'token': default_token_generator.make_token(user),
+        'uid': int_to_base36(user.pk),
         'LANGUAGE_CODE': user.primary_language,
     }
 
-    send_mail(
-        template_name='bb_accounts/activation_email',
-        subject=subject,
-        to=user,
-        **c
-    )
+    # If there is no password set then use the welcome + password set template
+    if not user.password:
```

Comment:
What will happen if someone logs in with Facebook? Will they get the _set your password_ mail?